client/__tests__/e2e/data.persistence.e2e.test.ts

```diff
@@ -0,0 +1,543 @@
+/**
+ * Data persistence smoke tests
+ * Tests that annotation changes persist across page refreshes by:
+ * 1. Performing annotation operations (create, edit, delete)
+ * 2. Refreshing the page
+ * 3. Verifying the changes persisted
+ *
+ * Each test uses a unique user ID to avoid conflicts when running in parallel
+ */
+
+/* eslint-disable compat/compat -- not ran in the browser */
+/* eslint-disable no-await-in-loop -- await in loop is needed to emulate sequential user actions  */
+import { test } from "@chromatic-com/playwright";
+import crypto from "crypto";
+import fs from "fs/promises";
+import path from "path";
+import { existsSync } from "fs";
+
+import mockSetup from "./playwright.global.setup";
+
+import {
+  calcDragCoordinates,
+  drag,
+  snapshotTestGraph,
+  goToPage,
+  createCategory,
+  addLabelToCategoryWithSelection,
+  assertCategoryExists,
+  assertCategoryDoesNotExist,
+  assertLabelExists,
+  assertLabelDoesNotExist,
+  deleteCategory,
+  renameCategory,
+  deleteLabel,
+  renameLabel,
+} from "./cellxgeneActions";
+import { APP_URL_BASE } from "../common/constants";
+
+const { describe } = test;
+
+// Test constants
+const TEST_CATEGORY_NAME = "persistence_test_category";
+const TEST_LABEL_NAME = "persistence_test_label";
+const SECOND_LABEL_NAME = "persistence_second_label";
+const RENAMED_CATEGORY_NAME = "renamed_persistence_category";
+const RENAMED_LABEL_NAME = "renamed_persistence_label";
+
+// Use persistence test URL with /w/ dataroot
+const DATASET = "pbmc3k.cxg";
+const persistenceTestURL = [APP_URL_BASE, "w", DATASET, ""].join("/");
+
+const PROJECT_ROOT =
+  process.env.PROJECT_ROOT ??
+  (existsSync(path.join(process.cwd(), "example-dataset"))
+    ? process.cwd()
+    : path.resolve(process.cwd(), ".."));
+
+// TODO #754
+test.beforeEach(mockSetup);
+
+const SIDE_PANEL = "layout-graph-side";
+
+const graphInstanceTestIds = ["layout-graph", "layout-graph-side"];
+
+// Helper function to generate unique user ID for each test
+function generateUniqueUserId(): string {
```

Comment:
Randomized `test-user-*` IDs are a good call to keep tiled user directories from colliding when we parallelize.